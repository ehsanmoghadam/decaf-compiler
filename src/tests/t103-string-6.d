int main() {
    string s;
    s = "He had just walked in. Hey, he said, \"Aren't you a string?\"";
    Print(s);
    s = "Anounymous answered: \'No I'm an single dummy backslash. :(\'";
    Print(s);
    s = "\"Oh! I am a single backslash too! I can be with you but I have a problem.\", he said.";
    Print(s);
    s = "\'What do you mean when you say \"problem\"?. We are just two backslashs like this:\\\\\', Anounymous said.";
    Print(s);
    s = "'I have a newline at my end. :| Because of this i can\'t be friend of any other backslashs (\\). In a single quote that is inside double quote i am like this: (\"\'\\\n\'\")', Poor Anounymous said.";
    Print(s);
    s = "\"No problem! I have a newline before. I can understand you. Let's be together!\", he said.";
    Print(s);
    s = "\"Ok! :) but in a double quote home.\", They found each other forever: \"\n\\\\\n\"";
    Print(s);
}