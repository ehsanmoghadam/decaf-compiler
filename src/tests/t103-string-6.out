He had just walked in. Hey, he said, "Aren't you a string?"
Anounymous answered: 'No I'm an single dummy backslash. :('
"Oh! I am a single backslash too! I can be with you but I have a problem.", he said.
'What do you mean when you say "problem"?. We are just two backslashs like this:\\', Anounymous said.
'I have a newline at my end. :| Because of this i can't be friend of any other backslashs (\). In a single quote that is inside double quote i am like this: ("'\
'")', Poor Anounymous said.
"No problem! I have a newline before. I can understand you. Let's be together!", he said.
"Ok! :) but in a double quote home.", They found each other forever: "
\\
"
